### PR TITLE
VALUES list

### PR DESCRIPTION
**Pull request by dimarub2000**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):
#5984 implemented
#6217 Fixed

### CODE REVIEW

File: dbms/src/TableFunctions/TableFunctionFactory.cpp [hunk 1]
```diff
@@ -17,11 +17,16 @@ namespace ErrorCodes
...
+void TableFunctionFactory::registerFunction(const std::string & name, Creator creator, CaseSensitiveness case_sensitiveness)
 {
-    if (!functions.emplace(name, std::move(creator)).second)
+    if (!table_functions.emplace(name, creator).second)
         throw Exception("TableFunctionFactory: the table function name '" + name + "' is not unique",
             ErrorCodes::LOGICAL_ERROR);
+
+    if (case_sensitiveness == CaseInsensitive
+        && !case_insensitive_table_functions.emplace(Poco::toLower(name), creator).second)
+        throw Exception("TableFunctionFactory: the case insensitive aggregate function name '" + name + "' is not unique",
```
Comment (alexey-milovidov) [inline dbms/src/TableFunctions/TableFunctionFactory.cpp]: Not an `aggregate function`

File: dbms/src/TableFunctions/TableFunctionValues.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        const auto & [value_field, value_type_ptr] = evaluateConstantExpression(args[i], context);
+        const TupleBackend & value_tuple = value_field.safeGet<Tuple>().toUnderType();
+
+        if (value_tuple.size() != sample_block.columns())
+            throw Exception("Values size should match with number of columns", ErrorCodes::LOGICAL_ERROR);
+
+        for (size_t j = 0; j < value_tuple.size(); ++j)
+        {
+            Field value = convertFieldToType(value_tuple[j], *sample_block.getByPosition(j).type, value_type_ptr.get());
```
Comment (alexey-milovidov) [inline dbms/src/TableFunctions/TableFunctionValues.cpp]: What if it cannot `convertFieldToType`?

Comment (dimarub2000) [reply dbms/src/TableFunctions/TableFunctionValues.cpp]: `Received exception from server (version 19.13.1):
Code: 53. DB::Exception: Received from localhost:9000. DB::Exception: Type mismatch in IN or VALUES section. Expected: String. Got: UInt64. `
Test will be added soon.

Comment (dimarub2000) [reply dbms/src/TableFunctions/TableFunctionValues.cpp]: > What if it cannot `convertFieldToType`?

For example : `Received exception from server (version 19.13.1):
Code: 53. DB::Exception: Received from localhost:9000. DB::Exception: Type mismatch in IN or VALUES section. Expected: String. Got: UInt64. `

Do we need to throw more local exception?
Tests will be added soon

**Comment by dimarub2000** [comment]
Single column ValuesList queries doesn't work yet. Trying to implement.

File: dbms/src/TableFunctions/TableFunctionValues.cpp (hunk 2 above)
Comment (alexey-milovidov) [reply dbms/src/TableFunctions/TableFunctionValues.cpp]: It's Ok.

File: dbms/src/TableFunctions/TableFunctionValues.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <TableFunctions/ITableFunction.h>
+
+namespace DB
+{
+
+class TableFunctionValues : public ITableFunction
```
Comment (alexey-milovidov) [inline dbms/src/TableFunctions/TableFunctionValues.h]: Comment needed.

File: dbms/src/Storages/StorageValues.h [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <ext/shared_ptr_helper.h>
+#include <Storages/IStorage.h>
+
+
+namespace DB
+{
+
+class StorageValues : public ext::shared_ptr_helper<StorageValues>, public IStorage
```
Comment (alexey-milovidov) [inline dbms/src/Storages/StorageValues.h]: Comment needed.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

**Comment by dimarub2000** [comment]
Latest commits also fix problem #6217, fix is implemented in `parseColumnsListFromString` function which is used in `ITableFunctionFileLike.cpp` and `TableFunctionValues.cpp`.

File: dbms/src/TableFunctions/parseColumnsListForTableFunction.h [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <Core/Block.h>
+
+
+namespace DB
+{
+
+void parseColumnsList(const std::string & structure, Block & sample_block, const Context & context);
```
Comment (alexey-milovidov) [inline dbms/src/TableFunctions/parseColumnsListForTableFunction.h]: Comment needed.
Better to return Block by value.

Comment (dimarub2000) [reply dbms/src/TableFunctions/parseColumnsListForTableFunction.h]: There is comment in later commit

**Comment by alexey-milovidov** [comment]
Continued in #6255

**Pull request merged**